week1/intro.java

```diff
@@ -0,0 +1,38 @@
+class Foo {
+    int order = 1; 
+    public Foo() {
+        
+    }
+
+    public void first(Runnable printFirst) throws InterruptedException {
+        
+        // printFirst.run() outputs "first". Do not change or remove this line.
+        printFirst.run();
+        order++;
+    }
```

Comment:
If `second` runs before `first`, would `second` sleep forever? Should `first` notify `second` here?

Reply:
probably. `third` would end up notifying `second` but if for some reason `third` is conditional on `second` then `second` will stall

Reply:
fixed